idpi/src/operators/flexpart.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+
+
+# similar to the subtract.accumulate but permute the order of the operans of the diff
+# TODO implement as a ufunc
+def cumdiff(A, axis):
+    r = np.empty(np.shape(A))
+    t = 0  # op = the ufunc being applied to A's  elements
+    for i in range(np.shape(A)[axis]):
+        t = np.take(A, i, axis) - t
+
+        slices = []
+        for dim in range(A.ndim):
+            if dim == axis:
+                slices.append(slice(i, i + 1))
+            else:
+                slices.append(slice(None))
+
+        r[tuple(slices)] = np.expand_dims(t, axis=t.ndim)
+    return r
+
+
+def fflexpart(ds, istep):
+    ds_out = {}
+    for field in (
+        "U",
+        "V",
+        "T",
+        "QV",
+        "PS",
+        "U_10M",
+        "V_10M",
+        "T_2M",
+        "TD_2M",
+        "CLCT",
+        "W_SNOW",
+    ):
+        ds_out[field] = ds[field].isel(step=istep)
+
+    ds_out["TOT_CON"] = (
+        (ds["TOT_CON"].isel(step=istep) - ds["TOT_CON"].isel(step=istep - 1))
```

Comment:
This factor accounts for the fact that FLEXPART expects precipitation rates per h. One should add a comment.